src/polygon.c

```diff
@@ -850,6 +852,54 @@ spherepoly_in(PG_FUNCTION_ARGS)
 	PG_RETURN_POINTER(poly);
 }
 
+Datum spherepoly_deg(PG_FUNCTION_ARGS)
+{
+	int num_elements, i, point_index, np;
+	ArrayType *float_vector;
+	float8 *array_data;
+	SPOLY *spoly;
+	SPoint *points;
+
+	float_vector = PG_GETARG_ARRAYTYPE_P(0);
+	num_elements = ArrayGetNItems(
+		ARR_NDIM(float_vector), ARR_DIMS(float_vector)
+	);
+	np = num_elements / 2;
+
+	if (num_elements <= 0)
+	{
+		PG_RETURN_NULL();
+	}
```

Comment:
Is not printing an error when passed an array with zero elements intentional and fitting with other pgSphere usage?

Reply:
So you didn't answer my question, but, based on your changes, I presume your answer was that an array with zero elements _should_ emit an error.

If that's the case, then you don't need these lines (869-874) at all. The test and error message are redundant with lines 875-882.